indexer/services/comlink/__tests__/controllers/api/v4/affiliates-controller.test.ts

```diff
@@ -171,34 +178,144 @@ describe('affiliates-controller#V4', () => {
   });
 
   describe('GET /snapshot', () => {
-    it('should return snapshots when all params specified', async () => {
+    const defaultInfo: AffiliateInfoCreateObject = testConstants.defaultAffiliateInfo;
+    const defaultInfo2: AffiliateInfoCreateObject = testConstants.defaultAffiliateInfo2;
+    const defaultInfo3: AffiliateInfoCreateObject = testConstants.defaultAffiliateInfo3;
+
+    beforeEach(async () => {
+      await testMocks.seedData();
+      // Create username for defaultWallet
+      await SubaccountUsernamesTable.create(testConstants.defaultSubaccountUsername);
+
+      // Create defaultWallet2, subaccount, and username
+      await WalletTable.create(testConstants.defaultWallet2);
+      await SubaccountTable.create(testConstants.defaultSubaccountDefaultWalletAddress);
+      await SubaccountUsernamesTable.create(
+        testConstants.subaccountUsernameWithDefaultWalletAddress,
+      );
+
+      // Create defaultWallet3, create subaccount, create username
+      await WalletTable.create(testConstants.defaultWallet3);
+      await SubaccountTable.create(testConstants.defaultSubaccountWithAlternateAddress);
+      await SubaccountUsernamesTable.create(testConstants.subaccountUsernameWithAlternativeAddress);
+
+      // Create affiliate infos
+      await AffiliateInfoTable.create(defaultInfo);
+      await AffiliateInfoTable.create(defaultInfo2);
+      await AffiliateInfoTable.create(defaultInfo3);
+    });
```

Comment:
nit: utilize await Promise.all() to parallelize what can be parallelized and speed up the test